### PR TITLE
Add create client page

### PR DESCRIPTION
**Pull request by hihuz**
In this PR I am adding the page to create clients as well as several components which will be reused for client updates as well.

Several things I wasn't sure of that would be nice to clarify (cc @johannespfeiffer):

* is it fine to implicitly create / update grant_types to always be `AUTHORIZATION_CODE` + `REFRESH_TOKEN`? or something else? Idea being that I save implementation of an UI component and simplify state tracking for the forms this way

* about scope, should I allow users to specify `offline` scope? (I have hidden it for now)

* about `redirect_uri`, should it be optional for users? (double checking)

* are there any specific combinations of scopes/grantTypes/redirect_uri/secret that should be forbidden? It should certainly be restricted on the backend if that is the case but having a UI which prevents form submission would also be nice in that case imo

* I haven't had time to implement any kind of UI notifications when queries / mutations are successful so I simply take user back to the main page with the client list after creation, is it fine for now?

Here are some screenshots:

regular screen:
Screenshot 2019-10-24 at 15 14 37

I am not very satisfied of the alignment of the checkboxes for scopes but couldn't come up with anything better 🤔any suggestions?
 
small screen: 
Screenshot 2019-10-24 at 15 20 15

### CODE REVIEW

File: console-src/src/components/ClientForm/ClientForm.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        <TextInput
+          label={copy.clientForm.name}
+          placeholder={copy.clientForm.placeholders.name}
+          value={name}
+          handleChange={this.handleTextInputChange("name")}
+        />
+        <CheckboxGroup title={copy.clientForm.scopes.title}>
+          {Object.keys(scopes).map(scope => (
+            <Checkbox
+              // @ts-ignore
```
Comment (hihuz) [inline console-src/src/components/ClientForm/ClientForm.tsx]: I had some nasty error here and couldn't figure out a fix 🤔 if no better I'll leave this ignore for now

Screenshot 2019-10-24 at 15 28 57

File: console-src/src/types/oAuthClient.tsx [hunk 2]
```diff
@@ -1,16 +1,16 @@
 enum Scope {
-  OFFLINE = "offline",
-  ACCOUNTS = "accounts",
-  USERS = "users",
-  TRANSACTIONS = "transactions",
-  TRANSFERS = "transfers",
-  SUBSCRIPTIONS = "subscriptions",
-  STATEMENTS = "statements"
+  OFFLINE = "OFFLINE",
```
Comment (hihuz) [inline console-src/src/types/oAuthClient.tsx:2]: we expose these as uppercase in our graph so lowercase was actually incorrect here

**Comment by johannespfeiffer** [comment]
> is it fine to implicitly create / update grant_types to always be `AUTHORIZATION_CODE` + `REFRESH_TOKEN`? or something else? Idea being that I save implementation of an UI component and simplify state tracking for the forms this way

Yes, only this should be exposed here at the moment.


> about scope, should I allow users to specify `offline` scope? (I have hidden it for now)

Yes, please allow `offline` - without it you cannot use `REFRESH_TOKEN` at all.
It won't work when you try to use the `offline` scope with PKCE (webapp only) which is already validated in the backend.


> about `redirect_uri`, should it be optional for users? (double checking)

Please make it mandatory. It is only null for grant type `password`, but required for `authorization_code`.


> are there any specific combinations of scopes/grantTypes/redirect_uri/secret that should be forbidden? It should certainly be restricted on the backend if that is the case but having a UI which prevents form submission would also be nice in that case imo

Not at this level. The client can decide if he wants to use plain `authorization_code` or `authorization_code` with PKCE extension. (The latter does not allow `offline` but requires no `secret`) - only if you would provide a selection of grant_types (see Q1) it would make to allow sth. like "Authorization Code (Backend)", "Authorization Code (Webapp)"


> * I haven't had time to implement any kind of UI notifications when queries / mutations are successful so I simply take user back to the main page with the client list after creation, is it fine for now?

✅KISS for the 1st iteration 😀

**Comment by hihuz** [comment]
Thanks a lot @johannespfeiffer 🙂 I have:

* made redirectUri mandatory
* exposed offline scope

Here is how it looks like now: 
Screenshot 2019-10-25 at 09 15 17

Screenshot 2019-10-25 at 09 15 38

File: console-src/src/types/oAuthClient.tsx (hunk 2 above)
Comment (johannespfeiffer) [reply console-src/src/types/oAuthClient.tsx:2]: You are right, `type-graphql` exposes the enum key - not its internal value.

File: console-src/src/components/ClientForm/ClientForm.tsx (hunk 1 above)
Comment (johannespfeiffer) [reply console-src/src/components/ClientForm/ClientForm.tsx]: Please see 744228a for a small suggestion.

**Review by johannespfeiffer — APPROVED** [review]

**Comment by hihuz** [comment]
Looks great 🙏 thank you very much

**Pull request merged**